File: algorithms/1_easy/T/01556_thousand_separator/solution.py
class Solution:
    def thousandSeparator(self, n: int) -> str:
        if n == 0:
            return "0"

        res = []
        dig_count = 0

        while n > 0:
            dig_count += 1
            res.append(chr(n % 10 + ord('0')))
            n //= 10

            if n > 0 and dig_count % 3 == 0:
                res.append('.')

        res.reverse()

        return "".join(res)

File: algorithms/1_easy/T/01556_thousand_separator/solution.cpp
class Solution {
public:
    string thousandSeparator(int n) {
        if (n == 0) {
            return "0";
        }

        string res{};
        res.reserve(13);
        int dig_count{0};

        while (n > 0) {
            dig_count += 1;
            res.push_back(static_cast<char>(n % 10) + '0');
            n /= 10;

            if (n > 0 && dig_count % 3 == 0) {
                res.push_back('.');
            }
        }

        reverse(res.begin(), res.end());

        return res;
    }
};
